src/session/session_api.c

```diff
@@ -49,6 +49,61 @@ __wt_session_reset_cursors(WT_SESSION_IMPL *session, bool free_buffers)
 	return (ret);
 }
 
+/*
+ * __wt_session_cursor_cache_sweep --
+ *	Sweep the cursor cache.
+ */
+int
+__wt_session_cursor_cache_sweep(WT_SESSION_IMPL *session)
+{
+	WT_CURSOR *cursor, *cursor_tmp;
+	WT_CURSOR_LIST *cached_list;
+	WT_DECL_RET;
+	uint32_t position;
+	int i, t_ret, nbuckets, nchecked, ncleaned;
+	bool caching;
+	bool productive;
+
+	position = session->cursor_sweep_position;
+	productive = true;
+	nbuckets = nchecked = ncleaned = 0;
+
+	caching = F_ISSET(session, WT_SESSION_CACHE_CURSORS);
```

Comment:
This needs a comment because I expected that if `caching` is false we'd return immediately.  A comment about why we cannot would be helpful here. My guess is that the session could have been reconfigured.

Reply:
You're right, we should return immediately. I've simplified the code.